lib/global_id/locator.rb

```diff
@@ -139,6 +126,26 @@ class ActiveRecordFinder
         def locate(gid)
           gid.model_class.find gid.model_id
         end
+
+        def locate_many(gids, options = {})
+          models_and_ids  = gids.collect { |gid| [ gid.model_class, gid.model_id ] }
+          ids_by_model    = models_and_ids.group_by(&:first)
+          loaded_by_model = Hash[ids_by_model.map { |model, ids|
+            [ model, find_records(model, ids.map(&:last), ignore_missing: options[:ignore_missing]).index_by { |record| record.id.to_s } ]
```

Comment:
Maybe what we really need to extract to the Active Record finder is the `find_records` method (and rename it `locate_many`. This way users who define their own locators only have to locate many models and all the other stuff is handled for them.